streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -0,0 +1,122 @@
+        memtableMinFlushTimeSensor = RocksDBMetrics.memtableMinFlushTimeSensor(streamsMetrics, metricContext);
+        memtableMaxFlushTimeSensor = RocksDBMetrics.memtableMaxFlushTimeSensor(streamsMetrics, metricContext);
+        writeStallDurationSensor = RocksDBMetrics.writeStallDurationSensor(streamsMetrics, metricContext);
+        blockCacheDataHitRatioSensor = RocksDBMetrics.blockCacheDataHitRatioSensor(streamsMetrics, metricContext);
+        blockCacheIndexHitRatioSensor = RocksDBMetrics.blockCacheIndexHitRatioSensor(streamsMetrics, metricContext);
+        blockCacheFilterHitRatioSensor = RocksDBMetrics.blockCacheFilterHitRatioSensor(streamsMetrics, metricContext);
+        bytesReadDuringCompactionSensor = RocksDBMetrics.bytesReadDuringCompactionSensor(streamsMetrics, metricContext);
+        bytesWrittenDuringCompactionSensor =
+            RocksDBMetrics.bytesWrittenDuringCompactionSensor(streamsMetrics, metricContext);
+        compactionTimeAvgSensor = RocksDBMetrics.compactionTimeAvgSensor(streamsMetrics, metricContext);
+        compactionTimeMinSensor = RocksDBMetrics.compactionTimeMinSensor(streamsMetrics, metricContext);
+        compactionTimeMaxSensor = RocksDBMetrics.compactionTimeMaxSensor(streamsMetrics, metricContext);
+        numberOfOpenFilesSensor = RocksDBMetrics.numberOfOpenFilesSensor(streamsMetrics, metricContext);
+        numberOfFileErrorsSensor = RocksDBMetrics.numberOfFileErrorsSensor(streamsMetrics, metricContext);
+    }
+
+    public void addStatistics(final String statisticsId,
+                              final Statistics statistics,
+                              final StreamsMetricsImpl streamsMetrics,
+                              final TaskId taskId) {
+        if (statisticsToRecord.isEmpty()) {
+            init(streamsMetrics, taskId);
+        }
+        final Statistics previousStatistics = statisticsToRecord.get(statisticsId);
+        if (previousStatistics != null) {
+            previousStatistics.close();
+        }
+        statisticsToRecord.put(statisticsId, statistics);
+    }
+
+    public void record() {
+        bytesWrittenToDatabaseSensor.record(0);
+        bytesReadToDatabaseSensor.record(0);
+        memtableBytesFlushedSensor.record(0);
+        memtableHitRatioSensor.record(0);
+        memtableAvgFlushTimeSensor.record(0);
+        memtableMinFlushTimeSensor.record(0);
+        memtableMaxFlushTimeSensor.record(0);
+        writeStallDurationSensor.record(0);
+        blockCacheDataHitRatioSensor.record(0);
+        blockCacheIndexHitRatioSensor.record(0);
+        blockCacheFilterHitRatioSensor.record(0);
+        bytesReadDuringCompactionSensor.record(0);
+        bytesWrittenDuringCompactionSensor.record(0);
+        compactionTimeAvgSensor.record(0);
+        compactionTimeMinSensor.record(0);
+        compactionTimeMaxSensor.record(0);
+        numberOfOpenFilesSensor.record(0);
+        numberOfFileErrorsSensor.record(0);
+    }
```

Comment:
The `record` method will change in the next PR when the metrics are computed. Currently, it only avoids compiler warnings.